### PR TITLE
WIP: Use pure go sqlite driver instead of cgo go-sqlite3 and remove go build tags related with sqlite

### PR DESCRIPTION
**Pull request by lunny**
This PR tries to use pure go sqlite driver instead of cgo go-sqlite3 for better compile time.

### CODE REVIEW

File: routers/init.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func initDBEngine(ctx context.Context) (err error) {
 	for i := 0; i < setting.Database.DBConnectRetries; i++ {
 		select {
 		case <-ctx.Done():
-			return fmt.Errorf("Aborted due to shutdown:\nin retry ORM engine initialization")
+			return fmt.Errorf("aborted due to shutdown:\nin retry ORM engine initialization")
```
Comment (delvh) [inline routers/init.go:72]: I have absolutely no clue where your hatred for capital letters comes from…
How have they traumatized you?

**Review by delvh — COMMENTED** [review]
I have two questions:
1) Why does this library bring **its own C library**?
2) How does adding **far** more code improve compile time?

File: routers/init.go (hunk 1 above)
Comment (wxiaoguang) [reply routers/init.go:72]: That's GO styling. GO standard says: every error message starts with lower-case. Please read the standards.

Comment (delvh) [reply routers/init.go:72]: Really? That's convention in Go?
Okay, then ignore my comment.

Comment (wxiaoguang) [reply routers/init.go:72]: Google it: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by lunny** [comment]
> I have two questions:
> 
> 1. Why does this library bring **its own C library**?
> 2. How does adding **far** more code improve compile time?

1. I think since the pure go sqlite library was translated from original sqlite, so the author also translate them. Maybe he will change them at some point.
2. This is a CGO problem. When compile with CGO, the Go compiler will spend more time. But when compile a pure Go project which even has more source files, it's faster.

**Comment by wxiaoguang** [comment]
After reading the code of `modernc.org/sqlite`, I'd like to say, it's surprisingly interesting and extremely impressive (well, I can not find other words to describe it because I only know a few words to describe an awesome thing ...)

The `modernc/sqlite` team wrote a C99 compiler frontend in Go, and wrote a lot of cross-platform code for their Go-based libc, then they compile the sqlite official code by their C compiler using their libc into Go, and embed it into the modernc/sqlite ....

ps: I haven't done any test and I am not sure how much time could be save by this Go-sqlite during Gitea compiling, just leave a comment about how it works.

**Comment by lunny** [comment]
> After reading the code of `modernc.org/sqlite`, I'd like to say, it's surprisingly interesting and extremely impressive (well, I can not find other words to describe it because I only know a few words to describe an awesome thing ...)
> 
> The `modernc/sqlite` team wrote a C99 compiler frontend in Go, and wrote a lot of cross-platform code for their Go-based libc, then they compile the sqlite official code by their C compiler using their libc into Go, and embed it into the modernc/sqlite ....
> 
> ps: I haven't done any test and I am not sure how much time could be save by this Go-sqlite during Gitea compiling, just leave a comment about how it works.

I think much time and memory in developer's machine and CI agents will be saved when CGO is gone. Because we can drop the xgo to do cross-compile. When using xgo, you have to check or download a very big image(about 5GB) and spent many time to compile because c++ compile is slower than pure Go.

**Pull request closed without merging**